pkg/kubectl/cmd/util/cmdtools.go

```diff
@@ -85,7 +85,7 @@ func defaultEnvShell() []string {
 	return []string{shell, flag}
 }
 
-func defaultEnv(cmdName CmdName, envs []string) ([]string, bool) {
+func defaultEnv(cmdName CmdName, envs []string, diffViewer string) ([]string, bool) {
```

Comment:
I think that's even worse
```
func Complete() {
	env = GetEnv()
	fvalue = GetFlagValue()
	if len(env) == 0 && len(fvalue) > 0 {
		o.DiffTool = fvalue
	} else if len(env) > 0 && len(fvalue) == 0 {
		o.DiffTool = env
	} else if env != fvalue {
		return err
	} else { // env and fvalue both is ""  {
		o.DiffTool = checkGitDiffInstalled(gitDiff, normalDiff)
	}

}

func RunApplyDiffApplied() {
	// launch the cmd tool using o.DiffTool
        // we have to directly use CmdTools here instead of NewDefaultCmdTool
        // becase edit command rely on it
	diff := CmdTool{
		Name:  DiffCmd,
		Args:  o.DiffTools,
		Shell: false,   //not sure if we can set hard-coded false here
                //bash -c "command_string" is only for test ? if not we also should move defaultEnvShell logic to here...
	}
	
	diff.LaunchTempFiles(tmpFileSlice...)
}
```